### PR TITLE
[WIP] Jabrefonline apollo

### PR DESCRIPTION
**Pull request by Siedlerchr**
- [ ] Change in `CHANGELOG.md` described in a way that is understandable for the average user (if applicable)
- [ ] Tests created for changes (if applicable)
- [ ] Manually tested changed features in running JabRef (always required)
- [ ] Screenshots added in PR description (for UI changes)
- [ ] [Checked documentation](https://docs.jabref.org/): Is the information available and up to date? If not created an issue at <https://github.com/JabRef/user-documentation/issues> or, even better, submitted a pull request to the documentation repository.

### CODE REVIEW

**Comment by Siedlerchr** [comment]
@tobiasdiez  I theoretically fixed all this modularity stuff, however, now I am having troubles with Kotlin shit
```

Users/christophs/workspace/jabref/build/generated/source/apollo/main/service/GetDocumentByIdQuery.java:277: error: cannot access Function1
        final GetUserDocumentRaw getUserDocumentRaw = reader.readObject($responseFields[0], new ResponseReader.ObjectReader<GetUserDocumentRaw>() {
                                                            ^
  class file for kotlin.jvm.functions.Function1 not found
/Users/christophs/workspace/jabref/build/generated/source/apollo/main/service/GetDocumentByIdQuery.java:342: error: cannot access Function2
          writer.writeList($responseFields[3], fields, new ResponseWriter.ListWriter() {
                ^

```

**Comment by Siedlerchr** [comment]
Screw this Apollo. Isn't there a better client?

**Comment by tobiasdiez** [comment]
Should work now...the kotlin reference is a module, so need to be required explicitly in the module spec

**Comment by Siedlerchr** [comment]
Ah yes thanks! compiles. Now I have to adapt the output package

**Comment by tobiasdiez** [comment]
https://github.com/apollographql/apollo-android/issues/2019

Probably just add the current path to the sourceset.

**Comment by Siedlerchr** [comment]
Now I only need to get it working in Eclipse^^
Eclipse doesn't like the patch modules somehow. Would be interesting if this works in Intellij.

**Comment by koppor** [comment]
As far as I understand the motivation, this makes JabRef a server for some online clients.

I would like to take this as a chance to work on a gradle multi module build (follow-up to https://github.com/JabRef/jabref/pull/3704).

**Comment by koppor** [comment]
This PR adds a new binary file:

grafik

I would like to see that file (`apollo-api-jvm-2.5.8.jar`) being added in a seperate module only - and keep the normal build "clean".

(I assume, the dependency isn't https://mvnrepository.com/artifact/com.apollographql.apollo/apollo-compiler/2.5.8)

**Comment by Siedlerchr** [comment]
@koppor  This is the dependency https://mvnrepository.com/artifact/com.apollographql.apollo/apollo-api-jvm
The problem is that runtime is a split package and we patch it with the api module

File: build.gradle [hunk 1]
```diff
@@ -64,6 +66,7 @@ application {
 modularity.patchModule("test", "fastparse_2.12-1.0.0.jar")
 modularity.patchModule("test2", "fastparse-utils_2.12-1.0.0.jar")
 modularity.patchModule("test3", "sourcecode_2.12-0.1.4.jar")
+modularity.patchModule("apollo.runtime", "apollo-api-jvm-2.5.8.jar")
```
Comment (Siedlerchr) [inline build.gradle:69]: @koppor

**Comment by tobiasdiez** [comment]
> As far as I understand the motivation, this makes JabRef a server for some online clients.

Nope, it makes JabRef online a possible server that JabRef can sync with (in addition to the Postgres, MySql, ... databases we already support).

File: build.gradle (hunk 1 above)
Comment (koppor) [reply build.gradle:69]: Please add a comment that if this is fixed, we can remove the `jar`

**Comment by koppor** [comment]
DevCall: Try it out in IntelliJ (and explain to koppor ^^)

**Comment by Siedlerchr** [comment]
./gradlew generateApolloSources

**Comment by Siedlerchr** [comment]
No matter what I try, I can't get the patched module stuff working in Eclipse.

**Comment by tobiasdiez** [comment]
So the imports are not working, or what is the problem?

**Comment by Siedlerchr** [comment]
As this is still heavy WIP I will move this to koppor to

**Comment by Siedlerchr** [comment]
Now located at https://github.com/koppor/jabref/pull/505

**Comment by tobiasdiez** [comment]
I would say we create a very minimal initial version that works, and then take this as the base branch for further PRs.

What is the problem with this branch though? Anything else that is not working than the eclipse integration?

**Comment by Siedlerchr** [comment]
The key problem is that a) Eclipse is not working with the patched module, it won't compile and can't find the module when it's located in runtime  and b) there was also a problem with the mutation code generation. Need to try this out again.

**Comment by Siedlerchr** [comment]
https://bugs.eclipse.org/bugs/show_bug.cgi?id=529923
https://wiki.eclipse.org/Java9/ModularityOptions

**Pull request closed without merging**